src/main/java/com/enterprise/fundraisermanager/controller/FundraiserManagerController.java

```diff
@@ -1,97 +1,125 @@
+        return "start";
+    }
+
+    /**
+     * Saves fundraiser to database.
+     * @param fundraiser
+     * @return start page
+     */
+    @RequestMapping("/saveFundraiser")
+    public String saveFundraiser(Fundraiser fundraiser){
+        try {
+            fundraiserService.save(fundraiser);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "start";
+        }
+        return "start";
+    }
+
+    /**
+     * Get all information regarding the fundraiser
+     * @param id
+     * @return all information regarding the fundraiser
+     */
+    @GetMapping("/fundraiser/{id}/")
+    public ResponseEntity fetchFundraiserByID(@PathVariable("id") Integer id) {
+        Fundraiser foundFundraiser = fundraiserService.fetchById(id);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity(foundFundraiser, headers, HttpStatus.OK);
+    }
+
+    /**
+     * Create a new fundraiser
+     * @param fundraiser
+     * @return new fundraiser object
+     */
+    @PostMapping(value="/fundraiser", consumes="application/json", produces = "application/json")
+    public Fundraiser createFundraiser(@RequestBody Fundraiser fundraiser) {
+        Fundraiser newFundraiser = null;
+        try {
+            newFundraiser = fundraiserService.save(fundraiser);
+        }catch (Exception e) {
+        }
+        return newFundraiser;
+    }
+
+    /**
+     * Delete a fundraiser
+     * @param id
+     * @return fundraisers that don't include the one that was removed
+     */
+    @DeleteMapping("/fundraiser/{id}/")
+    public ResponseEntity deleteFundraiser(@PathVariable("id") Integer id) {
+        try {
+            fundraiserService.delete(id);
+            return new ResponseEntity(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Search for a fundraiser.
+     * @param searchTerm
+     * @return all fundraisers similar to the search word that was typed into the search box.
+     */
+    @GetMapping(value="/fundraisers", consumes="application/json", produces="application/json")
+    public ResponseEntity searchFundraisers(@RequestParam(value = "searchTerm", required = false, defaultValue = "None") String searchTerm) {
+        List<Fundraiser>fundraisers= fundraiserService.fetchAll(searchTerm);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity(fundraisers, headers, HttpStatus.OK);
+
+    }
+    @GetMapping("/fundraisers")
+    public String searchFundraisersForm(@RequestParam(value="searchTerm", required=false, defaultValue="None")  String searchTerm, Model model) {
+        List<Fundraiser>fundraisers= fundraiserService.fetchAll(searchTerm);
+        model.addAttribute("fundraisers", fundraisers);
+        return "fundraisers";
```

Comment:
I modified the search feature so that it would generate a search API that would display all fundraisers that were similar to the search term entered into the search box. The ResponseEntity class represents the whole HTTP response, including the status code, headers, and body. As a result, we may fully configure the HTTP response using it.  

This is a helpful link: https://www.youtube.com/watch?v=xJa-UsP9slQ&ab_channel=BrandanJones